packages/logo-grid/partials/dialog-tooltip/index.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState, useRef, useEffect } from 'react'
+import { DialogOverlay, DialogContent } from '@reach/dialog'
+import VisuallyHidden from '@reach/visually-hidden'
+import { useRect } from '@reach/rect'
+import s from './style.module.css'
+import Portal from '@reach/portal'
+import InlineSvg from '@hashicorp/react-inline-svg'
+import svgX from '../../icons/x.svg?include'
+import classNames from 'classnames'
+
+interface DialogTooltipProps {
+  /** Elements to render in the content area of the dialog. */
+  children: React.ReactNode
+  /** Function to set the shown state of the dialog. Necessary for close functionality. */
+  setIsShown: React.Dispatch<React.SetStateAction<boolean>>
+  /** Whether to show the dialog or not. */
+  shown: boolean
+  /** DOMRect of the element that triggered the dialog. */
+  triggerRect: DOMRect
+  /** */
+  arrowSize?: number
+  /** Minimum distance in pixels that the dialog should be from the viewport edge.  */
+  collisionBuffer?: number
+  /** Color scheme appearance of the component. Works best in contexts with a matching theme. */
+  theme?: 'light' | 'dark'
+}
+
+function DialogTooltip({
```

Comment:
Similar to the Reach-based `tooltip` created for `tabs`, I split this out in the hopes that it might someday see re-use. I did not try to consolidate with the `tabs` tooltip - in this case, we have interactive content in our tooltip, so we need to use a more Dialog-like component.